internal/imex/imex.go

```diff
@@ -0,0 +1,108 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package imex
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/klog/v2"
+
+	spec "github.com/NVIDIA/k8s-device-plugin/api/config/v1"
+)
+
+// Channels represents a set of IMEX channels.
+type Channels []*channel
+
+// channel represents an IMEX channel.
+type channel struct {
+	id      string
+	devRoot string
+}
```

Comment:
This abstraction "feels" a little strange to me (in that when i just see this at the top of the file its not clear to me why the abstraction of a channel would have a `devRoot` in it). I get why you are doing it, and I don't have a better suggestion -- just thought it was worth pointing out the awkwardness of it.

Reply:
Could we alternatively just construct the name, path and hostpath once and store them here (rather than having helper functions for them)?